méli-mélo/2024-02-charts/js/charts.js

```diff
@@ -0,0 +1,165 @@
+			}
+
+			// Creating HTML nodes
+			figureElm = document.createElement( "figure" );
+			figcaptionElm = document.createElement( "figcaption" );
+			figcaptionElm.innerHTML = elm.caption.innerHTML;
+			figcaptionElm.classList.add( "h3" );
+			canvasElm = document.createElement( "canvas" );
+			canvasElm.setAttribute( "aria-label", elm.caption.innerHTML + i18nText.tableFollowing );
+			canvasElm.setAttribute( "role", "img" );
+			detailsElm = document.createElement( "details" );
+			detailsElm.classList.add( "mrgn-tp-md" );
+			summaryElm = document.createElement( "summary" );
+			summaryElm.innerHTML = elm.caption.innerHTML + i18nText.tableMention;
+
+			// Inserting HTML nodes
+			elm.parentNode.insertBefore( figureElm, elm );
+			figureElm.appendChild( figcaptionElm )
+			figureElm.appendChild( canvasElm );
+			figureElm.appendChild( detailsElm );
+			detailsElm.appendChild( summaryElm );
+			detailsElm.appendChild( elm );
+
+			// Set Chart options
+			chartType = elm.dataset.chartType ? elm.dataset.chartType : "line";
+			chartOpts = elm.dataset.chartOptions ? JSON.parse( elm.dataset.chartOptions ) : {};
+			chartData = elm.dataset.chartData ? JSON.parse( elm.dataset.chartData ) : generateData( elm );
+
+			// Create the Chart
+			chart = new Chart( canvasElm, {
+				type: chartType,
+				data: chartData,
+				options: chartOpts
+			} );
+
+			// Create MutationObserver to update graph whenever the table is updated
+			tableObserver = new MutationObserver( function() {
+				chart.data = elm.dataset.chartData ? JSON.parse( elm.dataset.chartData ) : generateData( elm );
+				chart.options = elm.dataset.chartOptions ? JSON.parse( elm.dataset.chartOptions ) : {};
+				chart.update();
+			} );
+
+			tableObserver.observe( elm, { attributes: true, subtree: true } );
+
+			// Identify that initialization has completed
+			wb.ready( $elm, componentName );
+		}
+	},
+
+	generateData = function( element ) {
+		let labels = [],
+			datasets = [],
+			dataObj = {},
+			columnIdx,
+			hasRowHeading = false,
+			tableHead = element.tHead.rows[ 0 ].cells,
+			tableRows = element.tBodies[ 0 ].rows;
+
+		// If the <table> has a TH element as the first element, populate labels
+		if ( tableRows[ 0 ].cells[ 0 ].nodeName == "TH" ) {
+			for ( let i = 0; i < tableRows.length; i++ ) {
+				let tableRowHeading = tableRows[ i ].cells[ 0 ];
+
+				if ( tableRowHeading.dataset.chartValue ) {
+					labels.push( tableRowHeading.dataset.chartValue );
+				} else {
+					labels.push( tableRowHeading.textContent );
+				}
+			}
+
+			dataObj.labels = labels;
+			hasRowHeading = true;
+		}
+
+		// Populate datasets
+		hasRowHeading ? columnIdx = 1 : columnIdx = 0;
+		for ( columnIdx; columnIdx < tableHead.length; columnIdx++ ) {
+			if ( !tableHead[ columnIdx ].dataset.chartSetIgnore == true ) {
+				let datasetObj = { label: "", data: [] },
+					datasetOpts = tableHead[ columnIdx ].dataset.chartSetOptions;
+
+				datasetObj.label = tableHead[ columnIdx ].textContent;
```

Comment:
Note/Comment: We can also find the cell header by using the DOM property `headers` on the `<td>` element - https://html.spec.whatwg.org/multipage/tables.html#the-td-element

Reply:
If I try this, it returns "". So it doesn't work.